pkg/sidecar/sidecar_test.go

```diff
@@ -0,0 +1,93 @@
+package sidecar
```

Comment:
This file is a little repetitive. Could switch to test by table, if it seems too verbose.